sympy/solvers/ode.py

```diff
@@ -1482,6 +1491,255 @@ def _test_term(coeff, order):
+    # We can extend it for 1F1 and 0F1 type also.
+    f = func
+    x = func.args[0]
+    df = f.diff(x)
+
+    # making given equation in normal form
+    from sympy.core.logic import fuzzy_not
+    I1 = factor(cancel(A.diff(x)/2 + A**2/4 - B))
+
+    # computing shifted invariant(J1) of the equation
+    J1 = factor(cancel(x**2*I1 + S(1)/4))
+    num, dem = J1.as_numer_denom()
+    num = powdenest(expand(num))
+    dem = powdenest(expand(dem))
+    pow_num = set()
+    pow_dem = set()
+    # this function will compute the different powers of variable(x) in J1.
+    # then it will help in finding value of k. k is power of x such that we can express
+    # J1 = x**k * J0(x**k) then all the powers in J0 become integers.
+    def _power_counting(num):
+        _pow = {0}
+        for val in num:
+            if val.has(x):
+                if isinstance(val, Pow) and val.as_base_exp()[0] == x:
+                    _pow.add(val.as_base_exp()[1])
+                elif val == x:
+                    _pow.add(val.as_base_exp()[1])
+                else:
+                    _pow.update(_power_counting(val.args))
+        return _pow
+
+    pow_num = _power_counting((num, ))
+    pow_dem = _power_counting((dem, ))
+    pow_dem.update(pow_num)
+
+    _pow = pow_dem
+    k = gcd(_pow)
+
+    # computing I0 of the given equation
+    I0 = powdenest(simplify(factor(((J1/k**2) - S(1)/4)/((x**k)**2))), force=True)
+    I0 = factor(cancel(powdenest(I0.subs(x, x**(S(1)/k)), force=True)))
+    num, dem = I0.as_numer_denom()
+
+    max_num_pow = max(_power_counting((num, )))
+    dem_args = dem.args
+    sing_point = []
+    dem_pow = []
+    # calculating singular point of I0.
+    for arg in dem_args:
+        if arg.has(x):
+            if isinstance(arg, Pow):
+                # (x-a)**n
+                dem_pow.append(arg.as_base_exp()[1])
+                sing_point.append(list(roots(arg.as_base_exp()[0], x).keys())[0])
+            else:
+                # (x-a) type
+                dem_pow.append(arg.as_base_exp()[1])
+                sing_point.append(list(roots(arg, x).keys())[0])
+
+    dem_pow.sort()
+    # checking if equivalence is exists or not.
+
+    if equivalence(max_num_pow, dem_pow) == "2F1":
+        return {'I0':I0, 'k':k, 'sing_point':sing_point, 'type':"2F1"}
+    else:
+        return None
+
+def ode_2nd_hypergeometric(eq, func, order, match):
+
+    from sympy.simplify.hyperexpand import hyperexpand
+    from sympy import factor
+    x = func.args[0]
+    f = func
+    C0, C1 = get_numbered_constants(eq, num=2)
+    a = match['a']
+    b = match['b']
+    c = match['c']
+
+    A = match['A']
+    B = match['B']
+
+    sol = None
+    if match['type'] == "2F1":
+        if c.is_integer == False:
+            sol = C0*hyper([a, b], [c], x) + C1*hyper([a-c+1, b-c+1], [2-c], x)*x**(1-c)
+        elif c == 1:
+            y2 = Integral(exp(Integral((-(a+b+1)*x + c)/(x**2-x), x))/(hyper([a, b], [c], x)**2), x)*hyper([a, b], [c], x)
```

Comment:
```
In [10]: Integral(sqrt(x - 1)/(x*hyper((1/2, -1), (1,), x)**2), x).doit()
Out[10]: 
⌠                         
⎮         _______         
⎮       ╲╱ x - 1          ``
⎮ ───────────────────── dx
⎮                     2   
⎮    ┌─  ⎛0.5, -1 │  ⎞    
⎮ x⋅ ├─  ⎜        │ x⎟    
⎮   2╵ 1 ⎝   1    │  ⎠    
⌡                       ``
```
`Integral `is not able to simplify hyper and it is returning unevaluated Integral. But it can be integrate by using `hyperexpand`.
```
In [12]: Integral(sqrt(x - 1)/(x*hyperexpand(hyper((1/2, -1), (1,), x))**2), x).
    ...: doit()
Out[12]: 
                                    _______
        ⎛  _______⎞   1.0⋅(2 - x)⋅╲╱ x - 1 
1.0⋅atan⎝╲╱ x - 1 ⎠ - ─────────────────────
                                 2         
                                x       

```
Due to this solution is coming with unevaluated integral. So I will use `hyperexpand`.

Reply:
That's fine. Maybe we can revisit this later - the paper had some suggestions for avoiding tricky integrals.